src/main/java/com/cloudbees/jenkins/plugins/mtslavescloud/MansionCloud.java

```diff
@@ -289,9 +293,68 @@ public Credential createAccessToken(URL broker) throws AbortException, OauthClie
         }
         label = Jenkins.getInstance().getLabel(st.getLabel()+" "+box.size+compat);
 
-        
+
         final Queue<PlannedMansionSlave> queue = new ArrayBlockingQueue<PlannedMansionSlave>(excessWorkload);
         List<PlannedNode> r = new ArrayList<PlannedNode>();
+
+        /**
+         * The current approach - fire up as much slaves as we can - is pretty broken.
```

Comment:
May be better. IMHO it's rather a documentation for the changelog or for JIRA issue. In the code it makes sense to clarify the current behavior and to reference issue/changelog with detailed description.